### DISCUSSION TITLE
Depends() not correctly generating openapi docs for list

### DISCUSSION THREAD

**Discussion by Adrian-at-CrimsonAzure** [fastapi/fastapi#8420] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Depends, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Test(BaseModel):
    id: int
    some_str: str
    list_values: list[str]


@app.get('/')
def test_get(test: Test = Depends()):
    return test
```

### Description

Generated doc has a body instead of a list query:
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/1719461/204891853-cc2f3ccf-0870-44d8-95ea-708a7acf4b85.png">

The same thing happens with:
```python
from fastapi import Depends, FastAPI

app = FastAPI()

def test_query(id: int, some_str: str, list_values: list[str]):
    return {id, some_str, list_values}


@app.get('/')
def test_get(test=Depends(test_query)):
    return test
```

Everything works correctly if I define the query params directly in the `test_get` method.

### Operating System

Linux

### Operating System Details

`python:3.11` Docker base image

### FastAPI Version

0.88.0, 0.86.0

### Python Version

3.11.0, 3.10.8

**Comment by mlbase** [comment]
@Adrian-at-CrimsonAuzre 
why are you using depends in pydantic model?
Depends function imply you will use cache memory for specific function or callable object. 
If you just want for typing and swagger-ui, remove it

**Comment by Adrian-at-CrimsonAzure** [comment]
> @Adrian-at-CrimsonAuzre why are you using depends in pydantic model? 

Because using a class with `Depends()` is a [documented](https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/) way of adding query parameters? The use of Pydantic classes specifically is also [commonly](https://stackoverflow.com/questions/67580885/fastapi-query-parameter-using-pydantic-model) [recommended](https://stackoverflow.com/questions/62468402/query-parameters-from-pydantic-model) [on](https://stackoverflow.com/questions/67814772/how-to-write-schema-for-query-parameters-in-fastapi-as-variable) [Stack Overflow](https://stackoverflow.com/questions/71177433/custom-pydantic-classes-as-fastapi-body-parameters). Abstraction of shared query parameters and the magic of Pydantic validators is nice. Plus it happens with also documented [method-based usage](https://fastapi.tiangolo.com/tutorial/dependencies/) of `Depends()`, so I'm reporting the bug.

**Comment by hellocoldworld** [comment]
Also running into this.

**Comment by EricKnowsCodeFu** [comment]
Bump please, also encountering this.

For an additional example, I would expect these two methods to produce the same interface, but it results in a GET endpoint that expects a body.

```
from fastapi import Depends, FastAPI, Query
from pydantic import BaseModel
from typing import List, Optional

app = FastAPI()


class CommonQueryParams(BaseModel):
    stuff: Optional[List[str]] = Query(default=[])


@app.get('/using-model')
async def using_model(
        query: CommonQueryParams = Depends(CommonQueryParams)
):
    pass


@app.get('/using-func')
async def using_func(
        stuff: Optional[List[str]] = Query(default=[])
):
    pass
```

The parameters of `using-func` is what I would expect `using-model` to have as well.

![image](https://user-images.githubusercontent.com/11470173/210422841-9a4fc891-0a4f-4686-ad13-349d8feb6110.png)
![image](https://user-images.githubusercontent.com/11470173/210422954-cc37a9bc-c449-408e-8f4a-4c845a012200.png)

**Comment by EricKnowsCodeFu** [comment]
For anyone who stumbles across this in the future, here's a solution adapted from this excellent Stack Overflow solution for similar problems with Form data: https://stackoverflow.com/a/60670614

```
def as_query(cls: Type[BaseModel]) -> Type[BaseModel]:
    """
    Coerces BaseModel fields into query string input fields
    """
    new_parameters = []

    for field_name, model_field in cls.__fields__.items():
        model_field: ModelField  # type: ignore

        new_parameters.append(
             inspect.Parameter(
                 model_field.alias,
                 inspect.Parameter.POSITIONAL_ONLY,
                 default=Query(...) if model_field.required else Query(model_field.default),
                 annotation=model_field.outer_type_,
             )
         )

    async def as_query_func(**data):
        return cls(**data)

    sig = inspect.signature(as_query_func)
    sig = sig.replace(parameters=new_parameters)
    as_query_func.__signature__ = sig  # type: ignore
    setattr(cls, 'as_query', as_query_func)
    return cls

annotated_params = as_query(CommonQueryParams)

@app.get('/using-annotated-model')
async def using_annotated_model(
        query: annotated_params = Depends(annotated_params.as_query)
):
    pass
```

**Comment by YuriiMotov** [comment, marked as answer]
Using Pydantic models to declare Query parameters [is supported since FastAPI 0.115.0](https://fastapi.tiangolo.com/tutorial/query-param-models/#query-parameters-with-a-pydantic-model) 🎉

Just replace `Depends` with `Query` to make it working:
```python
from fastapi import FastAPI, Query
from pydantic import BaseModel

app = FastAPI()


class Test(BaseModel):
    id: int
    some_str: str
    list_values: list[str]


@app.get('/')
def test_get(test: Test = Query()):
    return test

```
